govstack_api/building_blocks/bb_digital_registries/registries/insuree_registry.py

```diff
@@ -0,0 +1,209 @@
+        return registry_data
+
+    def read_record(self, mapped_data, only_first=True):
+        # it can be function above without common part
+        pass
+
+    def get_record_field(self, mapped_data, validate_data):
+        insuree_data = self.get_record(mapped_data, validate_data['field'])
+        insuree_data = self.change_result_extension(insuree_data, validate_data['ext'])
+        return insuree_data
+
+    def retrieve_filtered_records(self, mapped_data):
+        return self.get_record(mapped_data, only_first=False)
+
+    def update_registry_record(self, mapped_data):
+        insuree_uuid = self.get_record_field(self, mapped_data, "uuid", "string")
+        mapped_data['uuid'] = insuree_uuid
+        arguments_with_values = self.create_query_string(mapped_data)
+        query = get_update_registry_query(arguments_with_values)
+        self.client.execute_query(query)
+        return 200
+
+    def create_registry_record(self, mapped_data):
+        arguments_with_values = self.create_query_string(mapped_data)
+        query = get_create_registry_query(arguments_with_values)
+        self.client.execute_query(query)
+        return 200
+
+    def create_or_update_registry_record(self, mapped_data):
+        insuree_uuid = self.get_record_field(self, mapped_data, "uuid", "string")
+        if insuree_uuid:
+            self.update_registry_record(mapped_data)
+        else:
+            self.create_registry_record(mapped_data)
+
+    def map_to_graphql(self, validated_data):
+        mapped_data = {}
+        json_ext = {}
+
+        for http_field, value in validated_data.items():
+            if http_field in self.fields_mapping:
+                graphql_field = self.fields_mapping[http_field]
+                mapped_data[graphql_field] = value
+            elif http_field in self.special_fields:
+                json_ext[http_field] = value
+
+        if json_ext:
+            mapped_data['jsonExt'] = json.dumps(json_ext)
+
+        return mapped_data
+
+    def map_from_graphql(self, graphql_data):
+        mapped_data = {}
+        json_ext = {}
+
+        # We need to reverse the fields_mapping dictionary for map_from_graphql
+        reversed_fields_mapping = {v: k for k, v in self.fields_mapping.items()}
+
+        for graphql_field, value in graphql_data.items():
+            if graphql_field in reversed_fields_mapping:
+                http_field = reversed_fields_mapping[graphql_field]
+                mapped_data[http_field] = value
+            elif graphql_field == 'jsonExt':
+                if value is not None:
+                    json_ext = json.loads(value)
+                else:
+                    json_ext = {}
+
+        for special_field in ['BirthCertificateID', 'PersonalData']:
+            if special_field in json_ext:
+                mapped_data[special_field] = json_ext[special_field]
+
+        return mapped_data
+
+    def extract_records(self, result, only_first=True):
```

Comment:
those and methods below will be moved to helpes.py since they can be reusable with other models